extensions/assets-manager/Manifest.h

```diff
@@ -140,10 +140,21 @@ class CC_EX_DLL Manifest : public Ref
      */
     std::unordered_map<std::string, AssetDiff> genDiff(const Manifest *b) const;
     
+    /** @brief Generate difference between this Manifest and another, calling relevant callbacks on each asset.
+     * @param b           The other manifest
+     * @param onDiff      Callback for each changed asset, ignored if nullptr
+     * @param onUnchanged Callback for each unchanged asset, ignored if nullptr
+     */
+    void yieldDiff(const Manifest *b,
+                   const std::function<void(const std::string&, const AssetDiff&)>& onDiff,
+                   const std::function<void(const std::string&)>& onUnchanged) const;
+
     /** @brief Generate resuming download assets list
-     * @param units   The download units reference to be modified by the generation result
+     * @param units               The download units reference to be modified by the generation result
+     * @param getAssetStoragePath Function used to calculate the storage path of each asset
      */
-    void genResumeAssetsList(DownloadUnits *units) const;
+    void genResumeAssetsList(DownloadUnits *units,
+                             const std::function<std::string(const std::string& key, const Asset& asset)>& getAssetStoragePath) const;
```

Comment:
This is a method that is only used internally by AssetsManagerEx. With the default AssetsManagerEx::Config set, it will behave exactly the same as it used to be.

https://github.com/cocos2d/cocos2d-x/pull/15689/files#diff-3d58187924dcf35ee09bd7de35f8839fR59